tfx/components/example_gen/base_example_gen_executor_test.py

```diff
@@ -142,10 +143,97 @@ def testDoOutputSplit(self):
     # Check example gen outputs.
     self.assertTrue(tf.io.gfile.exists(self._train_output_file))
     self.assertTrue(tf.io.gfile.exists(self._eval_output_file))
+    
     # Output split ratio: train:eval=2:1.
     self.assertGreater(
         tf.io.gfile.GFile(self._train_output_file).size(),
         tf.io.gfile.GFile(self._eval_output_file).size())
+    
+  def testFeatureBasedPartition(self):
+    # Create exec proterties.
+    exec_properties = {
+        'input_config':
+            json_format.MessageToJson(
+                example_gen_pb2.Input(splits=[
+                    example_gen_pb2.Input.Split(
+                        name='single', pattern='single/*'),
+                ]),
+                preserving_proto_field_name=True),
```

Comment:
you can move input_config to setup() to reduce duplication.

just update output_config key in each test method